File: day-1/go/solution-b.go
package main

import (
	"fmt"
	"math"
	"os"
	"strconv"
	"strings"
)

func main() {
	contents, err := os.ReadFile("../input.txt")
	if err != nil {
		panic(err)
	}

	lines := strings.Split(string(contents), "\n")

	increments := 0
	lastWindow := math.MaxInt

	for i := 0; i < len(lines); i++ {
		if len(lines) < i+3 {
			break
		}
		window := 0
		for _, line := range lines[i : i+3] {
			number, err := strconv.Atoi(line)
			if err != nil {
				panic(err)
			}
			window += number
		}
		if lastWindow < window {
			increments++
		}
		lastWindow = window
	}

	fmt.Println("Solution:", increments)
}

File: day-1/typescript/src/solution-b.ts
export const solution = "b";

const input = await Bun.file("../input.txt").text();

const numbers = input.split("\n").map(Number);

let increments = 0;
let lastWindow = Infinity;

for (let i = 0; i < numbers.length; i++) {
  if (numbers.slice(i, i + 3).length < 3) break;
  const window = numbers.slice(i, i + 3).reduce((a, b) => a + b, 0);
  if (lastWindow < window) {
    increments++;
  }
  lastWindow = window;
}

console.log('Solution:', increments);